File: app/src/main/java/com/master/kit/testcase/banner/BannerActivity.java
package com.master.kit.testcase.banner;

import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;

import com.master.kit.R;
import com.master.kit.testcase.retrofit.IETongDaiService;
import com.master.kit.engine.imageloader.ImageLoaderFactory;
import com.master.kit.testcase.retrofit.ServiceHolder;
import com.ooftf.banner.Banner;
import com.ooftf.banner.BannerPagerAdapter;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import tf.oof.com.service.base.BaseSlidingActivity;


public class BannerActivity extends BaseSlidingActivity {
    private static final String TGA =  Thread.currentThread().getStackTrace()[1].getClassName();
    @BindView(R.id.banner)
    Banner banner;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_banner);
        ButterKnife.bind(this);
        initBanner();
    }

    private void initBanner() {
        IETongDaiService api = ServiceHolder.INSTANCE.getService();
        Call<BannerBean> call = api.getBanner("1", "2");
        call.enqueue(new Callback<BannerBean>() {
            @Override
            public void onResponse(Call<BannerBean> call, Response<BannerBean> response) {
                BannerBean result = response.body();
                List<BannerBean.BodyEntity.PicListEntity> picList = result.getBody().getPicList();
                List<String> list = new ArrayList<>();
                for(BannerBean.BodyEntity.PicListEntity e:picList){
                    list.add(e.getPicUrl());
                }
                banner.setUris(list, new BannerPagerAdapter.DisplayImageCallback() {
                    @Override
                    public void displayImage(String url, ImageView view) {
                        ImageLoaderFactory.create().display(BannerActivity.this,url,view);
                    }
                });

            }

            @Override
            public void onFailure(Call<BannerBean> call, Throwable t) {
                log(t.getMessage());
            }
        });
    }
    private void log(String result) {
        if (result == null)
            return;
        int step = 3000;
        for (int i = 0; i < result.length(); i = i + step) {
            if (i + step >= result.length()) {
                Log.e(TGA,"请求服务器返回数据 " + i / step + " : "
                        + result.substring(i, result.length()));
            } else {
                Log.e(TGA,"请求服务器返回数据 " + i / step + ": "
                        + result.substring(i, i + step));
            }

        }

    }

    @Override
    protected void onStart() {
        super.onStart();
        banner.stopCycle();
    }

    @Override
    protected void onStop() {
        super.onStop();
        banner.stopCycle();
    }
}

File: app/src/main/java/com/master/kit/testcase/banner/BannerActivity.kt
package com.master.kit.testcase.banner

import android.content.Context
import android.os.Bundle
import android.widget.ImageView
import com.master.kit.R
import com.master.kit.activity.widget.GuideActivity
import com.master.kit.dagger.DaggerBannerComponent
import com.master.kit.net.etd.PresenterObserver
import com.master.kit.net.etd.ResponseView
import com.master.kit.net.ServiceHolder
import com.trello.rxlifecycle2.kotlin.bindToLifecycle
import com.youth.banner.loader.ImageLoaderInterface
import io.reactivex.android.schedulers.AndroidSchedulers
import kotlinx.android.synthetic.main.activity_banner.*
import tf.oof.com.service.base.BaseSlidingActivity
import tf.oof.com.service.engine.image_loader.IImageLoader
import javax.inject.Inject


class BannerActivity : BaseSlidingActivity() {

    @Inject lateinit var imageLoader: IImageLoader
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        DaggerBannerComponent.create().inject(this)
        setContentView(R.layout.activity_banner)
        setupBanner()
        requestBanner()
        responseLayout.setOnRetryListener { requestBanner() }
        next.setOnClickListener {
            startActivity(GuideActivity::class.java)
        }
    }

    private fun setupBanner() {
        banner.setImageLoader(object : ImageLoaderInterface<ImageView> {
            override fun createImageView(context: Context): ImageView? {
                return null;
            }

            override fun displayImage(context: Context, path: Any, imageView: ImageView) {
                val bean = path as BannerBean.BodyEntity.PicListEntity
                imageLoader.display(context, bean.picUrl, imageView)
            }

        })
    }

    private fun requestBanner() {
        ServiceHolder
                .service
                .getBanner("1", "2")
                .bindToLifecycle(banner)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(HomeObserver(responseLayout))
    }

    inner class HomeObserver(responseView: ResponseView<BannerBean>) : PresenterObserver<BannerBean>(responseView) {
        override fun onResponseSuccess(bean: BannerBean) {
            super.onResponseSuccess(bean)
            banner.setImages(bean.body.picList)
            banner.start()
        }
    }
}
